### ISSUE TITLE
[Breaking Change] Missing label in spite of label-adding rule `XmsEnumChanged` failing

### ISSUE THREAD

**Issue by konrad-jamrozik** [Azure/azure-sdk-tools#6999]
- https://github.com/Azure/azure-rest-api-specs-pr/pull/14195
  - Informed by email from Jefferey R.
  - Notes in OneNote, in `Breaking change missing label 9/14`

**Comment by konrad-jamrozik** [comment]
PRs:
- [Pull Request 499784](https://devdiv.visualstudio.com/DevDiv/_git/openapi-alps/pullrequest/499784): Improve logging around reading breaking changes rules + fix typos in some names.

**Comment by konrad-jamrozik** [comment]
I tried to reproduce the "no label was added even through XmsEnumChanged rule was violated" problem present in this PR:
- https://github.com/Azure/azure-rest-api-specs-pr/pull/14195

on the TEST environment (test-repo-billy), but did not manage to successfully reproduce. The label was added as it should have. However, with the following two PRs deployed root-causing the problem should be easier, in case it happens again:

- [Pull Request 499784](https://devdiv.visualstudio.com/DevDiv/_git/openapi-alps/pullrequest/499784): Improve logging around reading breaking changes rules + fix typos in some names.
- [Pull Request 500076](https://devdiv.visualstudio.com/DevDiv/_git/openapi-alps/pullrequest/500076): Remove obsolete BreakingChangeRules.yml subset duplicate
